### PR TITLE
[testing] feat: matrix tests to check roles for wildcards

### LINKED ISSUE #8517: Minimize RBAC permissions of platform components
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

Kubernetes CIS [we recommend not using wildcard rights in Role/ClusterRole](https://github.com/aquasecurity/kube-bench/blob/0f8dfaf1156652b556c913aa8e78fccbb1dfd186/cfg/rke-cis-1.24/policies.yaml#L29-L35). 
However, the DKP platform does not take this recommendation into account everywhere. 
For example:
- dex has [verbs: *](https://github.com/deckhouse/deckhouse/blob/main/modules/150-user-authn/templates/dex/rbac-for-us.yaml#L53)
- upmeter-agent [has wildcard rights to all resources](https://github.com/deckhouse/deckhouse/blob/main/modules/500-upmeter/templates/upmeter-agent/rbac-for-us.yaml#L63-L73) in d8-upmeter namespace

Therefore, the DKP does not fully comply with the Kubernetes CIS recommendations.

### Proposed Solution

Grant specific required permissions to objects in the Kubernetes API.

### Additional Information

You might be interested in the [audit2rbac](https://github.com/liggitt/audit2rbac) tool to minimize permissions for the modules.

### PR DESCRIPTION
**Pull request by juev**
## Description

It is necessary to check for the presence of wildcards in Roles and ClusterRoles in the rbac-for-us.yaml files.

## Why do we need it, and what problem does it solve?

Kubernetes CIS [we recommend not using wildcard rights in Role/ClusterRole](https://github.com/aquasecurity/kube-bench/blob/0f8dfaf1156652b556c913aa8e78fccbb1dfd186/cfg/rke-cis-1.24/policies.yaml#L29-L35).

## What is the expected result?

With each commit, template files are checked for the presence of wildcards. If no exception rule has been added, a validation error is generated.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: testing
type: feature
summary: matrix tests to check roles for wildcards
impact_level: default
```

Closed #8517

### CODE REVIEW

File: testing/matrix/linter/rules/roles/wildcards.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+			}
+		}
+	}
+
+	converter := runtime.DefaultUnstructuredConverter
+
+	role := new(rbac.Role)
+	err := converter.FromUnstructured(object.Unstructured.UnstructuredContent(), role)
+	if err != nil {
+		panic(err)
```
Comment (juev) [inline testing/matrix/linter/rules/roles/wildcards.go:67]: Here I took a sample from other linters. But in fact, we can return an error.

File: testing/matrix/linter/rules/roles/wildcards.go [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+	for _, rule := range role.Rules {
+		var objs []string
+		if slices.Contains(rule.APIGroups, "*") {
+			objs = append(objs, "apiGroups")
+		}
+		if slices.Contains(rule.Resources, "*") {
+			objs = append(objs, "resources")
+		}
+		if slices.Contains(rule.Verbs, "*") {
+			objs = append(objs, "verbs")
```
Comment (yalosev) [inline testing/matrix/linter/rules/roles/wildcards.go:164]: Let's separate the errors, please. We can give an advice to a user, how to fix the verbs wildcard
Something like
```
"Verb contains the wildcard. Please, use explicitly set list: `"create", "delete", "deletecollection", "get", "list", "patch", "update", "watch"`
```

File: testing/matrix/linter/rules/roles/wildcards.go [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+		}
+		if len(objs) > 0 {
+			return errors.NewLintRuleError(
+				"WILDCARD001",
+				object.Identity(),
+				object.Path,
+				strings.Join(objs, ", ")+" contains a wildcards",
```
Comment (yalosev) [inline testing/matrix/linter/rules/roles/wildcards.go:171]:
```suggestion
				strings.Join(objs, ", ")+" contains a wildcards. Replace them with an explicit list of resources",
```

File: testing/matrix/linter/rules/roles/wildcards.go (hunk 2 above)
Comment (juev) [reply testing/matrix/linter/rules/roles/wildcards.go:164]: We can have wildcards in APIGroups, Resources and Verbs. If we separate the errors for each of these groups, should we provide information on the possible values ​​for each of the groups? Or will we do this only for Verbs?
Now we will have only one error for one role indicating in which groups we have wildcards.
If we divide errors into groups, we will have several times more errors. Is it worth doing?

**Review by yalosev — APPROVED** [review]

**Pull request merged**